src/test/kotlin/com/terraformation/backend/accelerator/db/ReportStoreTest.kt

```diff
@@ -693,6 +694,289 @@ class ReportStoreTest : DatabaseTest(), RunsAsDatabaseUser {
     }
   }
 
+  @Nested
+  inner class FetchOne {
+    @Test
+    fun `throws exception if user is not an organization manager, or global role user`() {
+      deleteOrganizationUser(organizationId = organizationId)
+      deleteUserGlobalRole(role = GlobalRole.AcceleratorAdmin)
+
+      insertProjectReportConfig()
+      val reportId =
+          insertReport(
+              quarter = ReportQuarter.Q4,
+              startDate = LocalDate.of(2030, Month.OCTOBER, 1),
+              endDate = LocalDate.of(2030, Month.DECEMBER, 31))
+
+      assertThrows<ReportNotFoundException> { store.fetchOne(reportId) }
+
+      insertOrganizationUser(role = Role.Contributor)
+      assertThrows<ReportNotFoundException> { store.fetchOne(reportId) }
+
+      deleteOrganizationUser(organizationId = organizationId)
+      insertOrganizationUser(role = Role.Manager)
+      assertDoesNotThrow { store.fetchOne(reportId) }
+
+      deleteOrganizationUser(organizationId = organizationId)
+      insertUserGlobalRole(role = GlobalRole.ReadOnly)
+      assertDoesNotThrow { store.fetchOne(reportId) }
```

Comment:
These assertions could use messages to say which case is failing; otherwise you have to go look at the test source and match up the line numbers.